pd-client/client.go

```diff
@@ -43,77 +50,271 @@ type Client interface {
 	Close()
 }
 
+type tsoRequest struct {
+	done     chan error
+	physical int64
+	logical  int64
+}
+
+const (
+	pdTimeout             = 3 * time.Second
+	maxMergeTSORequests   = 10000
+	maxInitClusterRetries = 100
+)
+
 type client struct {
-	worker *rpcWorker
+	urls        []string
+	clusterID   uint64
+	tsoRequests chan *tsoRequest
+
+	mu            sync.RWMutex
+	clients       map[string]pdpb2.PDClient
+	leader        string
+	checkLeaderCh chan struct{}
+
+	wg   sync.WaitGroup
+	quit chan struct{}
```

Comment:
Seem that etcd uses the context as the root context, I think it is ok.

We must use Context here to avoid gRPC using context.Background directly. If you don't like struct field, please pass Context to each client function, but if you think the context is the root context for the client, it is ok to be here.